plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -31,38 +50,58 @@ import java.time.Duration
 import java.time.Instant
 import kotlin.io.path.relativeTo
 
-sealed class CodeScanSessionConfig(
+class CodeScanSessionConfig(
     private val selectedFile: VirtualFile,
     private val project: Project,
     private val scanType: SecurityScanType
 ) {
     var projectRoot = project.guessProjectDir() ?: error("Cannot guess base directory for project ${project.name}")
         private set
 
-    abstract val sourceExt: List<String>
-
     private var isProjectTruncated = false
 
+    private val parsedGitIgnorePatterns = parseGitIgnore()
+
     /**
      * Timeout for the overall job - "Run Security Scan".
      */
-    abstract fun overallJobTimeoutInSeconds(): Long
+    private val language = selectedFile.programmingLanguage().toTelemetryType()
+
+    private val languageConfig: Map<CodewhispererLanguage, Pair<Long, Int>> = mapOf(
```

Comment:
We don't need this, the timeouts should be the same regardless of what language the active file is.

Project scans: 10 minutes, 5GB
File scans: 60 seconds, 200KB

Reply:
So just use default instead of this. will update in the new commit.